src/controllers/company.controller.js

```diff
@@ -0,0 +1,166 @@
+import crypto from "crypto";
+import bcrypt from "bcryptjs";
+import jwt from "jsonwebtoken";
+import GenerateResponse from "../utils/response.creator.js"
+import * as customerServices from "../services/customer.service.js";
+
+const RegisterCompany = async (req, res) => {
+    try {
+        const newRegistration = {
+            companyName : req.body.companyName,
+            companyId : req.body.companyId,
+            email : req.body.email,
+            password : req.body.password,
+            address : req.body.address,
+            registrationNumber : req.body.registrationNumber,
+            phoneNumber : req.body.phoneNumber,
+            ceoName : req.body.ceoName,
+            ceoContactNumber : req.body.ceoContactNumber,
+            ceoEmail : req.body.ceoEmail,
+            ceoAadhaarNumber : req.body.ceoAadhaarNumber,
+            verificationToken : crypto.randomBytes(32).toString("hex"),
+            verificationTokenExpires : Date.now() + 3600000,
+        };
+    
+        const existingCompany = await customerServices.FetchCompanyById(newRegistration.companyId);
+    
+        if(existingCompany){
+            return GenerateResponse(
+                res, 
+                400,
+                {},
+                "The Registration already Exist"
+            );
+        };
+    
+        const token = newRegistration.verificationToken;
+
+        newRegistration.password = await bcrypt.hash(newRegistration.password, parseInt(process.env.SALT));
+
+        console.log(newRegistration.password);
+    
+        await customerServices.CreateCompany(newRegistration);
+    
+        return GenerateResponse(
+            res,
+            200,
+            {token},
+            "The Company has been enlisted for our services please proceed to filling the form",
+        );
+    } catch (error) {
+        if(process.env.DEBUG == "TRUE") console.log(error);
+        if (
+            error.name == "ValidationError" ||
+            error.name == "CastError" ||
+            error.name == "BSONTypeError"
+        ) {
+            return GenerateResponse(
+                res,
+                400,
+                {},
+                "One or more inputs are invalid"
+            );
+        }
+        return GenerateResponse(res, 500);
+    }
+}
+
+const VerifyCompany = async(req, res) => {
+    try {
+        const verifyRegistration = {
+            verificationToken: req.body.verificationToken,
+            verifictionTokenExpires: { $gt: Date.now() },
```

Comment:
Where is this verifictionTokenExpires being used?

Reply:
to authenticate the company